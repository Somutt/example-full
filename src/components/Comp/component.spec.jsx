import { screen } from '@storybook/testing-library';
import { renderTheme } from '../../utils/renderTheme';
import { Comp } from '.';

describe('<Comp />', () => {
  it('Should render Comp correctly with title', () => {
    renderTheme(<Comp title='oi' />);
    const heading = screen.getByRole('heading', { name: /oi/i });
    expect(heading).toBeInTheDocument();
  });
  it('Should render Comp correctly without title', () => {
    renderTheme(<Comp />);
    const heading = screen.queryByRole('heading', { name: /oi/i });
    expect(heading).not.toBeInTheDocument();
  });
  it('Should match snapshot', () => {
    const { container } = renderTheme(<Comp title='ola' />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
