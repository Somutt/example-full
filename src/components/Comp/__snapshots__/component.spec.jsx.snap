// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comp /> Should match snapshot 1`] = `
<div
  class=""
>
  <h1>
    ola
  </h1>
</div>
`;
